pkg/settlement/swap/chequebook/chequebook.go

```diff
@@ -165,11 +199,20 @@ func lastIssuedChequeKey(beneficiary common.Address) string {
 
 // Issue issues a new cheque and passes it to sendChequeFunc
 // if sendChequeFunc succeeds the cheque is considered sent and saved
-func (s *service) Issue(beneficiary common.Address, amount *big.Int, sendChequeFunc SendChequeFunc) error {
+func (s *service) Issue(ctx context.Context, beneficiary common.Address, amount *big.Int, sendChequeFunc SendChequeFunc) error {
 	// don't allow concurrent issuing of cheques
-	// this would be sufficient on a per beneficiary basis
 	s.lock.Lock()
 	defer s.lock.Unlock()
+
+	availableBalance, err := s.AvailableBalance(ctx)
+	if err != nil {
+		return err
+	}
+
+	if amount.Cmp(availableBalance) > 0 {
```

Comment:
How is this error propagated in the stack?

In the ideal world, we should make it clear to the user that he should refund his chequebook.

Reply:
to accounting, where there will be a settlement failed log. then attempted overdrafts